pkg/adaptation/adaptation.go

```diff
@@ -588,12 +633,25 @@ func (r *Adaptation) sortPlugins() {
 	sort.Slice(r.plugins, func(i, j int) bool {
 		return r.plugins[i].idx < r.plugins[j].idx
 	})
+	sort.Slice(r.validators, func(i, j int) bool {
+		return r.validators[i].idx < r.validators[j].idx
+	})
 	if len(r.plugins) > 0 {
 		log.Infof(noCtx, "plugin invocation order")
 		for i, p := range r.plugins {
 			log.Infof(noCtx, "  #%d: %q (%s)", i+1, p.name(), p.qualifiedName())
 		}
 	}
+	if len(r.validators) > 0 {
+		log.Infof(noCtx, "validator plugin invocation order")
```

Comment:
Bringing over https://github.com/containerd/nri/pull/163#discussion_r2087686495

> I thought we were planning to invoke validators in parallel? (I don't think that has to be part of _this_ PR, but I think we shouldn't promise a specific invocation order. Maybe we shouldn't sort these at all, or explicitly randomly sort them?)

Reply:
True, we did. I updated the PR to run validation in parallel.

One more related thing I need to check though: is it goroutine-safe to access the same request locally in the builtin plugin and at the same time pass it over ttrpc to an external plugin from another goroutine. My assumption is that protoimpl.{SizeCache,MessageState} are set/used on the receiver side only, but this needs to be verified.

Reply:
Are there any places where we might need to update comments to explain that validation happens in parallel? (Just wondering if customers might try creating plugins with low `plugin_idx` and wonder why they don't receive priority)

Reply:
Yes, I think it will be a good idea to emphasize in the documentation that during the validation phase there is no 'plugin ordering by index' as validating plugins are run in parallel.

But regarding users wondering "why they don't get priority", I think they should not. Any presumed ordering/priority would be relevant only if it was possible to force validation both to fail and to succeed. But our semantics is deliberately such that the latter is not possible.